encodings/zstd/src/array.rs

```diff
@@ -88,11 +94,46 @@ fn choose_max_dict_size(uncompressed_size: usize) -> usize {
     (uncompressed_size / 100).clamp(256, 100 * 1024)
 }
 
-fn collect_valid(parray: &PrimitiveArray) -> VortexResult<PrimitiveArray> {
+fn collect_valid_primitive(parray: &PrimitiveArray) -> VortexResult<PrimitiveArray> {
     let mask = parray.validity_mask()?;
     filter(&parray.to_array(), &mask)?.to_primitive()
 }
 
+fn collect_valid_vbv(vbv: &VarBinViewArray) -> VortexResult<(Buffer<u8>, Vec<usize>)> {
+    let mask = vbv.validity_mask()?;
+    let mut buffer = BufferMut::empty();
+    let mut value_byte_indices = Vec::new();
+    for i in 0..vbv.len() {
+        if mask.value(i) {
+            value_byte_indices.push(buffer.len());
+            // here's where we write the string lengths
+            let value = vbv.bytes_at(i);
```

Comment:
Similarly, there's an array accessor trait to iterate bytes from the varbinview

Reply:
What trait are you referring to? I don't see it

Reply:
Ah never mind, found it.